### DISCUSSION TITLE
MeadiatR architecture best practice

### DISCUSSION THREAD

**Discussion by LukasKuchta** [LuckyPennySoftware/MediatR#789] [General]
HI, @jbogard, i read your advice why do not send another comman from handler. Better is create notification and consume it. 

Is there any hint what system commponent could be command/handler and what service method? Can i use cqrs only for db IO and others requirements make as methods in service? Where are the bounds

1) create customer command
2) publish event
3) send emal - question : should be done by 

- a) SendEmailCommand - bad violation your advice
- b) emailService.send()? Based on your advice b) is correct, right?

**Comment by eskaufel** [comment]
I would create a notification called UserCreated that is published in the CreateUser handler after the user is Created. Then have a NotificationHandler (called Email, Communication, or something like that) that handles UserCreated and sends an email. If you have other chores that should run after a user is created, just add more handlers.
